core/transactions/attempt_context_impl.cxx

```diff
@@ -3533,10 +3533,20 @@ attempt_context_impl::get_doc(const core::document_id& id,
 
   try {
     if (allow_replica) {
-      core::operations::lookup_in_any_replica_request req{ id };
-      req.read_preference = couchbase::read_preference::selected_server_group_or_all_available;
-      req.specs = specs;
-      execute_lookup(this, req, cb);
+      cluster_ref().with_bucket_configuration(
+        id.bucket(),
+        [this, id, specs, cb = std::move(cb)](
+          std::error_code ec, const std::shared_ptr<topology::configuration>& config) {
+          if (ec) {
+            cb(FAIL_OTHER, std::nullopt, ec.message(), std::nullopt);
```

Comment:
NIT: Maybe we could add something to the error message so it's clearer that we failed here:


```suggestion
            cb(FAIL_OTHER, std::nullopt, "failed to check whether access_deleted is supported: " + ec.message(), std::nullopt);
```